panels/applications/cc-applications-panel.c

```diff
@@ -1403,6 +1408,7 @@ populate_applications (CcApplicationsPanel *self)
 
   container_remove_all (GTK_CONTAINER (self->sidebar_listbox));
 
+  g_signal_handler_block (self->manager, self->app_filter_id);
```

Comment:
Why block this signal handler here? Do you have reason to believe that the main context might iterate while we’re inside this function, and hence end up with nested `populate_applications()` calls?

Reply:
Yes, that's precisely is my line of thought. If `app-filter-changed` emits at very short intervals (I don't know if this happens in real-life scenario), it might end up with long tail of `populate_applications()`.

Reply:
OK. I don’t think that’s possible, but blocking/unblocking the signal doesn’t hurt.